dsp-shared/src/test/scala/dsp/valueobjects/LangStringSpec.scala

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package dsp.valueobjects
+
+import dsp.errors.BadRequestException
+import dsp.errors.ValidationException
+import dsp.valueobjects.User._
+import zio.prelude.Validation
+import zio.test._
+
+/**
+ * This spec is used to test the [[dsp.valueobjects.LangString]] value objects creation.
+ */
+object LangStringSpec extends ZIOSpecDefault {
+  private val validLanguageCode   = "de"
+  private val invalidLanguageCode = "00"
+
+  def spec = (languageCodeTest + langStringTest)
+
+  private val languageCodeTest = suite("LanguageCode")(
+    test("pass an empty value and return an error") {
+      assertTrue(
+        LanguageCode.make("") == Validation.fail(ValidationException(LanguageCodeErrorMessages.LanguageCodeMissing))
+      )
+    },
+    test("pass an invalid value and return an error") {
+      assertTrue(
+        LanguageCode.make(invalidLanguageCode) == Validation.fail(
+          ValidationException(LanguageCodeErrorMessages.LanguageCodeInvalid)
+        )
+      )
+    },
+    test("pass a valid value and successfully create value object") {
+      assertTrue(LanguageCode.make(validLanguageCode).toOption.get.value == validLanguageCode)
+    },
+    suite("default language codes should be correct")(
+      test("English") {
+        val lang = "en"
+        (for {
+          manual <- LanguageCode.make(lang)
+          default = LanguageCode.en
+        } yield assertTrue(manual == default) &&
+          assertTrue(default.value == lang)).toZIO
```

Comment:
Do you need this `.toZIO`?

Reply:
yes, otherwise I had a weird error about missing implicites

Reply:
Ah yes, now I remember that I had the same problem when working on the user.

Reply:
yes... I copied your solution :P